src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -169,6 +197,47 @@ class thread_priority {
   long task_priority() const { return std::numeric_limits<long>::max() - (task_id + 1); }
 };
 
+/**
+ * Holds metrics for a given task/thread about retry counts and times. It is here
+ * because the mapping between tasks and threads can be complicated and can span
+ * different time ranges too.
+ */
+class task_metrics {
+  public:
```

Comment:
If no private then just use `struct`:
```suggestion
struct task_metrics {
```